includes/UserMailer.php

```diff
@@ -434,95 +433,154 @@ protected static function quotedPrintableCallback( $matches ) {
  *
  */
 class EmailNotification {
-	protected $action;
-	protected $subject, $body, $replyto, $from, $bodyHTML;
-	protected $timestamp, $summary, $minorEdit, $oldid, $composed_common;
-	protected $mailTargets = array();
+
+	/**
+	 * @var User
+	 */
+	private $editor;
 
 	/**
 	 * @var Title
 	 */
-	protected $title;
+	private $title;
+
+	private $timestamp;
+
+	private $summary;
+
+	private $minorEdit;
+
+	private $currentRevId;
+
+	private $previousRevId;
+
+	private $action;
+
+	private $otherParam;
+
+	private $subject;
+
+	private $body;
+
+	private $replyto;
+
+	private $from;
+
+	private $bodyHTML;
+
+	private $composedCommon = false;
 
 	/**
-	 * @var User
+	* @param User $editor
+	* @param Title $title
+	* @param string $timestamp : Edit timestamp
+	* @param string $summary : Edit summary
+	* @param bool $minorEdit
+	* @param int $currentRevId : Revision ID
+	* @param int $previousRevId : Revision ID
+	* @param string $action (Wikia)
+	* @param array $otherParam
 	 */
-	protected $editor;
+	public function __construct( $editor, $title, $timestamp, $summary, $minorEdit, $currentRevId = 0, $previousRevId = 0, $action = '', $otherParam = [] ) {
```

Comment:
This is a great candidate for the Builder pattern. There may be a slight upfront cost, but the advantages are maintainability, clarity for client and readability of the client usage.
If there are more parameters being added to this, it'll just get more error prone and harder to maintain.